### PR TITLE
Allow running many runs per test without flooding the logs

### PR DESCRIPTION
**Pull request by jtattermusch**
Based on #9161 (fix needed to prevent tests from hanging if they timeout).

Allow invoking run_tests_matrix.py   with specified number of runs per tests. To prevent the test reports from becoming too large (and to consume too much memory in a run_tests.py instance), also allow omitting passing tests from the XML report and from console output.

-- one more thing that needs to be addressed is increasing the timeout in run_tests_matrix.py accordingly (currently 4 hours should be enough for most wrapped languages, but it might not be enough for C/C++).

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -1097,6 +1094,12 @@ def runs_per_test_type(arg_str):
         help='Generates a JUnit-compatible XML report')
 argp.add_argument('--report_suite_name', default='tests', type=str,
         help='Test suite name to use in generated JUnit XML report')
+argp.add_argument('--quiet_success',
```
Comment (adelez) [inline tools/run_tests/run_tests.py]: Looks like this option only makes sense when n > 1, maybe we don't need to expose it as an arg?

File: tools/run_tests/run_tests.py [hunk 2]
```diff
@@ -779,7 +776,7 @@ def configure(self, config, args):
   def test_specs(self):
     return [
         self.config.job_spec(['src/objective-c/tests/run_tests.sh'],
-                              timeout_seconds=None,
+                              timeout_seconds=60*60,
```
Comment (adelez) [inline tools/run_tests/run_tests.py]: Define a constant for timeout values?

File: tools/run_tests/run_tests.py [hunk 3]
```diff
@@ -824,7 +821,7 @@ def test_specs(self):
     import yaml
     with open('tools/run_tests/sanity/sanity_tests.yaml', 'r') as f:
       return [self.config.job_spec(cmd['script'].split(),
-                                   timeout_seconds=None, environ={'TEST': 'true'},
+                                   timeout_seconds=30*60, environ={'TEST': 'true'},
```
Comment (adelez) [inline tools/run_tests/run_tests.py]: Ditto

**Comment by jtattermusch** [comment]
@adelez, I merged the PR before I received the comment from you.

Currently I am not planning to introduce this for interop tests because it will take more work to make use of changes this PR (and that's what I am planning to work on in the nearest future). We might try to introduce a similar change for interop tests later (but on the other hand, running multiple instances of interop tests is in fact what stress tests are doing).

**Pull request merged**